src/main/java/gregtech/common/EventHandlers.java

```diff
@@ -194,6 +194,12 @@ public static void onLivingEquipmentChangeEvent(LivingEquipmentChangeEvent event
                 valueItem.isItemEqual(MetaItems.QUANTUM_CHESTPLATE_ADVANCED.getStackForm())) {
             event.getEntity().isImmuneToFire = false;
         }
+
+        // Workaround to recipe caching issue with fluid jetpack
+        // TODO, rewrite logic and remove in armor rewrite
+        if (valueItem.isItemEqual(MetaItems.SEMIFLUID_JETPACK.getStackForm())) {
```

Comment:
Could prefer to use another method that compares the valueitems directly, instead of making a new ItemStack.